src/ControlzEx/Controls/TabControlEx.cs

```diff
@@ -31,34 +31,72 @@ namespace ControlzEx.Controls
 
+        /// <summary>Identifies the <see cref="TabPanelVisibility"/> dependency property.</summary>
         public static readonly DependencyProperty TabPanelVisibilityProperty =
             DependencyProperty.Register(nameof(TabPanelVisibility), typeof(Visibility), typeof(TabControlEx), new PropertyMetadata(Visibility.Visible));
 
         public static readonly DependencyProperty OwningTabItemProperty = DependencyProperty.RegisterAttached("OwningTabItem", typeof(TabItem), typeof(TabControlEx), new PropertyMetadata(default(TabItem)));
 
+        /// <summary>Helper for getting <see cref="OwningTabItemProperty"/> from <paramref name="element"/>.</summary>
+        /// <param name="element"><see cref="DependencyObject"/> to read <see cref="OwningTabItemProperty"/> from.</param>
+        /// <returns>OwningTabItem property value.</returns>
         [AttachedPropertyBrowsableForType(typeof(ContentPresenter))]
         public static TabItem GetOwningTabItem(DependencyObject element)
         {
             return (TabItem)element.GetValue(OwningTabItemProperty);
         }
 
+        /// <summary>Helper for setting <see cref="OwningTabItemProperty"/> on <paramref name="element"/>.</summary>
+        /// <param name="element"><see cref="DependencyObject"/> to set <see cref="OwningTabItemProperty"/> on.</param>
+        /// <param name="value">OwningTabItem property value.</param>
         public static void SetOwningTabItem(DependencyObject element, TabItem value)
         {
             element.SetValue(OwningTabItemProperty, value);
         }
 
+        public static readonly DependencyProperty OwningItemProperty = DependencyProperty.RegisterAttached("OwningItem", typeof(object), typeof(TabControlEx), new PropertyMetadata(default(object)));
+
+        /// <summary>Helper for setting <see cref="OwningItemProperty"/> on <paramref name="element"/>.</summary>
+        /// <param name="element"><see cref="DependencyObject"/> to set <see cref="OwningItemProperty"/> on.</param>
+        /// <param name="value">OwningItem property value.</param>
+        public static void SetOwningItem(DependencyObject element, object value)
+        {
+            element.SetValue(OwningItemProperty, value);
+        }
+
+        /// <summary>Helper for getting <see cref="OwningItemProperty"/> from <paramref name="element"/>.</summary>
+        /// <param name="element"><see cref="DependencyObject"/> to read <see cref="OwningItemProperty"/> from.</param>
+        /// <returns>OwningItem property value.</returns>
+        [AttachedPropertyBrowsableForType(typeof(ContentPresenter))]
+        public static object GetOwningItem(DependencyObject element)
+        {
+            return (object)element.GetValue(OwningItemProperty);
+        }
+
+        /// <summary>Identifies the <see cref="MoveFocusToContentWhenSelectionChanges"/> dependency property.</summary>
```

Comment:
It feels like that the OwningItem and OwningTabItem should be better readonly DependencyProperty to prevent unwanted changes from any user to it.

Reply:
We would have to do the same for `OwningTabItem` then (to be consistent) but that would be a breaking change...